mkvtoolnix is not building on Darwin

### Describe the bug
mkvtoolnix fails to build on Darwin

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix-env -i mkvtoolnix` on a Darwin system

### Expected behavior
mkvtoolnix builds and installs

### Additional context
Build error output:

```
checking whether the C compiler works... no
configure: error: in `/private/tmp/nix-build-mkvtoolnix-59.0.0.drv-0/source':
configure: error: C compiler cannot create executables
See `config.log' for more details
builder for '/nix/store/njbd7480qwzhsy1sigjlfmwz37n28ns7-mkvtoolnix-59.0.0.drv' failed with exit code 77
```

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@codyopel @rnhmjoj 

### Metadata

The problem seems to have popped up with 98dcfcfeb80477315f7faa995d677f919500f8b5

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.5.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.15`
 - channels(paikens): `"home-manager, nixos-21.05.2001.d4590d21006, nixpkgs-21.11pre306170.4d3e13e51b6, nixpkgs-master, nur-duckpuppy, nur-gerschtli, unstable-21.11pre306003.c464dc811ba, nixpkgs-unstable-21.11pre299378.db6e089456c, darwin"`
 - nixpkgs: `/nix/store/r2x4g0hd7cmbpqhy0ayclr89bz0ccg9x-nixpkgs-unstable.tar.gz`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/video/mkvtoolnix/default.nix
+++ b/pkgs/applications/video/mkvtoolnix/default.nix
@@ -1,6 +1,5 @@
 { lib
 , stdenv
-, mkDerivation
 , fetchFromGitLab
 , pkg-config
 , autoreconfHook
@@ -46,15 +45,15 @@ let
     '';
 
 in
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "mkvtoolnix";
-  version = "59.0.0";
+  version = "60.0.0";
 
   src = fetchFromGitLab {
     owner = "mbunkus";
     repo = "mkvtoolnix";
     rev = "release-${version}";
-    sha256 = "sha256-bPypOsveXrkz1V961b9GTJKFdgru/kcW15z/yik/4yQ=";
+    sha256 = "sha256-WtEC/EH0G1Tm6OK6hmVRzloLkO8mxxOYYZY7k/Wi2zE=";
   };
 
   nativeBuildInputs = [

